stdlib/public/SwiftShims/swift/shims/CMakeLists.txt

```
@@ -63,6 +63,31 @@ foreach(input ${sources})
     list(APPEND outputs "${output_dir_static}/${input}")
   endif()
 endforeach()
+if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
```

I suggest to gate this (and the similar checks below) behind a feature flag that by default is true on Darwin systems -- this way we can disable this in presets if this is causes any issues 
(Also it will serve to document the purpose of the code in the enclosed blocks, and to have a single place in which to evaluate the default)

One possible way to do this

```suggestion
if (SWIFT_STDLIB_INSTALL_PARENT_MODULE_FOR_SHIMS)
```

Definition of the flag in `stdlib/cmake/modules/StdlibOptions.cmake`
```
if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
  set(SWIFT_STDLIB_INSTALL_PARENT_MODULE_FOR_SHIMS_default TRUE)
else()
  set(SWIFT_STDLIB_INSTALL_PARENT_MODULE_FOR_SHIMS_default FALSE)
endif()

option(SWIFT_STDLIB_INSTALL_PARENT_MODULE_FOR_SHIMS
       "Install a parent module map for Swift shims."
       ${SWIFT_STDLIB_INSTALL_PARENT_MODULE_FOR_SHIMS_default})
```

---

Done. Thanks for providing actually usable text and not just "TODO".